src/unicorn_baseline/language/main.py

```diff
@@ -12,6 +12,200 @@
+        interval (int): Interval between checks in seconds.
+    """
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        for folder in runpath.iterdir():
+            if task_name in folder.name:
+                print(f"Predictions found in {folder}. Proceeding to postprocess.")
+                return folder
+        print("Waiting for predictions to complete...")
+        time.sleep(interval)
+    raise TimeoutError(f"Predictions not found within {timeout} seconds.")
+
+
+def drop_keys_except(data: List, keys: List[str]) -> List:
+    """
+    Drop all keys from the dictionary except the specified keys.
+    """
+    return [
+        {key: value for key, value in example.items() if key in keys}
+        for example in data
+    ]
+
+
+def post_process_predictions(data: json, task_config):
+    task_id = task_config.jobid
+    prediction_name = task_config.target.prediction_name
+    if prediction_name in (
+        "single_label_binary_classification",
+        "single_label_regression",
+    ):
+        for example in data:
+            example[prediction_name] = float(example[prediction_name])
+    if task_id == 15:
+        for example in data:
+            example[prediction_name] = [
+                example.pop("left"),
+                example.pop("right"),
+            ]
+    elif task_id == 16:
+        for example in data:
+            keys = [
+                "biopsy",
+                "cancer",
+                "high_grade_dysplasia",
+                "hyperplastic_polyps",
+                "low_grade_dysplasia",
+                "non_informative",
+                "serrated_polyps",
+            ]
+            example[prediction_name] = [
+                1.0 if example.pop(key) in ["True", True] else 0.0 for key in keys
+            ]
+    elif task_id == 17:
+        for example in data:
+            example[prediction_name] = [
+                float(example.pop("lesion_1")),
+                float(example.pop("lesion_2")),
+                float(example.pop("lesion_3")),
+                float(example.pop("lesion_4")),
+                float(example.pop("lesion_5")),
+            ]
+    data = drop_keys_except(data, ["uid", prediction_name])
+    return data
+
 
 def run_language():
-    pass
+    # Read the task configuration, few-shots and test data
+    # We'll leverage the DRAGON baseline algorithm for this
+    algorithm = DragonBaseline()
+    algorithm.load()
+    algorithm.analyze()  # needed for verification of predictions
+    task_config = algorithm.task
+    few_shots = algorithm.df_train
+    test_data = algorithm.df_test
+    basepath = Path("/opt/app/unicorn_baseline/src/unicorn_baseline/language/")
+    setup_folder_structure(basepath, test_data)
+    print(f"Task description: {task_config}")
+
+    task_name = task_config.task_name
+    task_id = task_config.jobid
+
+    # Task specific preprocessing
+    if "Task16_" in task_name:
```

Comment:
Just for safety, I would make it as strict as possible:
`if task_name.startswith("Task16_"):`